pkg/deploy/registry/deploylog/rest.go

```diff
@@ -0,0 +1,259 @@
+	namespace, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return c.rn.ReplicationControllers(namespace).Watch(lSelector, fSelector, rv)
+}
+
+type podGetter struct {
+	podsNamespacer kclient.PodsNamespacer
+}
+
+// Get is responsible for retrieving the deployer pod
+func (g *podGetter) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	namespace, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return g.podsNamespacer.Pods(namespace).Get(name)
+}
+
+// REST is an implementation of RESTStorage for the api server.
+type REST struct {
+	Deploy         *Client
+	PodGetter      pod.ResourceGetter
+	ConnectionInfo kclient.ConnectionInfoGetter
+	Timeout        time.Duration
+}
+
+// NewREST creates a new REST for DeploymentLogs. It uses three clients: one for configs,
+// one for deployments (replication controllers) and one for pods to get the necessary
+// attributes to assemble the URL to which the request shall be redirected in order to
+// get the deployment logs.
+func NewREST(
+	dn client.DeploymentConfigsNamespacer,
+	rn kclient.ReplicationControllersNamespacer,
+	pn kclient.PodsNamespacer,
+	connectionInfo kclient.ConnectionInfoGetter) *REST {
+
+	return &REST{
+		Deploy:         &Client{rn: rn, dn: dn},
+		PodGetter:      &podGetter{pn},
+		ConnectionInfo: connectionInfo,
+		Timeout:        defaultTimeout,
+	}
+}
+
+// NewGetOptions returns a new options object for deployment logs
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &deployapi.DeploymentLogOptions{}, false, ""
+}
+
+// New creates an empty DeploymentLog resource
+func (r *REST) New() runtime.Object {
+	return &deployapi.DeploymentLog{}
+}
+
+// Get returns a streamer resource with the contents of the deployment log
+func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	deployLogOpts, ok := opts.(*deployapi.DeploymentLogOptions)
+	if !ok {
+		return nil, errors.NewBadRequest("did not get an expected options.")
+	}
+
+	// Fetch deploymentConfig
+	config, err := r.Deploy.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("deploymentConfig", name)
+	}
+	// See latest version; if 0, there are no deployments for this config
+	desiredVersion := config.LatestVersion
+	if desiredVersion == 0 {
+		return nil, errors.NewBadRequest(fmt.Sprintf("no deployment exists for deploymentConfig %q", config.Name))
+	}
+
+	// Support retrieving logs for older deployments
+	switch {
+	case deployLogOpts.Version == -1:
+		// Latest
+	case deployLogOpts.Version <= 0 || deployLogOpts.Version > config.LatestVersion:
+		// Invalid version
+		return nil, errors.NewBadRequest(fmt.Sprintf("invalid version for deploymentConfig %q: %d", config.Name, deployLogOpts.Version))
+	default:
+		desiredVersion = deployLogOpts.Version
+	}
+
+	// List deployments
```

Comment:
All I meant was that you just have to make the deployment name and do a `Get` rather than searching sorted list results. If the user doesn't supply a version, the logs you want are the deployment named `util.LatestDeploymentNameForConfig`. If the user supplies a version, they want the deployment named `{configName}-{version}` (could be another util to construct the name). Then you just `Get(name)`.